### PR TITLE
refactor: merge Hey logic onto Peermngr

### LINKED ISSUE #119: refactor: merge hey service with peer manager
Attempt to simplify the codebase by merging the logic of hey.go with peermngr.go

### PR DESCRIPTION
**Pull request by gallexis**
closes #119

### CODE REVIEW

File: exchange/peermgr.go [hunk 1]
```diff
@@ -63,11 +64,86 @@ func NewPeerMgr(h host.Host, regions []Region) *PeerMgr {
...
+		_, err := s.Write(buf)
+		if err != nil {
+			log.Error().Err(err).Msg("could not write bytes")
+		}
+	}()
+}
+
 // Receive a new greeting from peer
-func (pm *PeerMgr) Receive(p peer.ID, h Hey) {
+func (pm *PeerMgr) receive(p peer.ID, h Hey) {
```
Comment (tchardin) [inline exchange/peermgr.go:146]:
```suggestion
func (pm *PeerMgr) receiveHey(p peer.ID, h Hey) {
```
Might be more consistent

**Review by tchardin — APPROVED** [review]
Don't have any strong opinions on this. If you think it makes things simpler, I trust your point of view. Thanks

File: exchange/peermgr.go (hunk 1 above)
Comment (gallexis) [reply exchange/peermgr.go:146]: Hum, how about handleHey instead? Because we already have a getHey

Comment (tchardin) [reply exchange/peermgr.go]: receiveHey isn't the same as getHey though

Comment (gallexis) [reply exchange/peermgr.go]: ah yes true but I though it was confusing

**Pull request merged**